### PR TITLE
docs: update usage of `track` to new API

### PR DESCRIPTION
**Pull request by avivr**
# What is it?

- [ ] Feature / enhancement
- [ ] Bug
- [x] Docs / tests

# Description

Update example usage of `track` to the new non-deprecated API.

# Use cases and why

So people will learn the new syntax

# Checklist:

- [x] My code follows the [developer guidelines of this project](../CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] Added new tests to cover the fix / functionality

### CODE REVIEW

File: packages/docs/src/routes/examples/apps/reactivity/auto-complete/app.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ export const AutoComplete = component$(() => {
   });
 
   useWatch$(async ({ track }) => {
-    const searchInput = track(state, 'searchInput');
+    const searchInput = track(state.searchInput);
```
Comment (shairez) [inline packages/docs/src/routes/examples/apps/reactivity/auto-complete/app.tsx:32]: should it be wrapped with a function as well?

**Comment by shairez** [comment]
Thanks @avivr 

Added a comment

File: packages/docs/src/routes/examples/apps/reactivity/auto-complete/app.tsx (hunk 1 above)
Comment (shairez) [reply packages/docs/src/routes/examples/apps/reactivity/auto-complete/app.tsx]: thanks!

**Comment by shairez** [comment]
Thanks for the fixes @avivr !

Please fix the merge conflicts and push again

**Comment by manucorporat** [comment]
Please revert all the changes inside the starters folder, they are e2e tests, we cant fully change it all to the new version since the old way still works and should work!

**Comment by manucorporat** [comment]
Amazing work!! thanks a lot for the PR!

**Pull request merged**